### PR TITLE
[ADD] new module hr_timesheet_task_change_project

### PR DESCRIPTION
**Pull request by huguesdk**

### CODE REVIEW

File: hr_timesheet_task_change_project/models/account_analytic_line.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        pass
+
+    def onchange_project_id(self):
+        # in the hr_timesheet module, this method sets task_id to False if the
+        # selected project is not the task's project. this override removes
+        # this behavior.
+        task_id = self.task_id
+        # the parent method returns a domain to limit the available tasks, so
+        # it needs to be returned.
+        result = super()
```
Comment (huguesdk) [inline hr_timesheet_task_change_project/models/account_analytic_line.py]: this seems to work, but shouldn’t it be:
```suggestion
        result = super().onchange_project_id()
```

Comment (victor-champonnois) [reply hr_timesheet_task_change_project/models/account_analytic_line.py]: En effet, j'ai essayé de comprendre pourquoi super() marchait tout seul, mais sans succès. Je dirai que ce serait une bonne idée d'utiliser `super().onchange_project_id()` car c'est ce qui est fait partout.

**Review by victor-champonnois — APPROVED** [review]
LGTM :+1: ( même si c'est compliqué de vérifier toutes les conséquences que cette PR peut avoir.)

**Comment by huguesdk** [comment]
actually, none of the methods in `AccountAnalyticLine` were called, and it seemed to work because the task domain is also [defined in the view](https://github.com/coopiteasy/OCB/blob/7d5207b4fc41d876faf917a233b3ca8cb2542af5/addons/hr_timesheet/views/hr_timesheet_views.xml#L46). they weren’t called because the decorators were missing. the decorators add attributes on the methods they decorate, and this is how odoo finds them back. simply redefining methods in sub-classes results in new methods without these attributes, so odoo doesn’t find them and thus doesn’t call them. fixed.

**Pull request merged**